### PR TITLE
Field overrides: skipping overrides for properties no longer existing in plugin

### PR DESCRIPTION
**Pull request by mckn**
**What this PR does / why we need it**:
Prior to this PR the field overrides editor would explode if trying to render an override containing properties that does not longer exists in the panel.

This PR prevents that from happening by rendering the properties as "unkown property" and adds a possibility to remove it from the configuration.

Skärmavbild 2021-01-12 kl  07 29 33

### CODE REVIEW

**Comment by mckn** [comment]
I'm just going to add some tests before moving this from draft.

File: public/app/features/dashboard/components/PanelEditor/OverrideEditor.tsx [hunk 1]
```diff
@@ -114,8 +103,13 @@ export const OverrideEditor: React.FC<OverrideEditorProps> = ({
     });
 
   const renderOverrideTitle = (isExpanded: boolean) => {
-    const overriddenProperites = override.properties.map(p => registry.get(p.id).name).join(', ');
+    const overriddenProperites = override.properties
+      .map(p => registry.getIfExists(p.id)?.name)
+      .filter(name => !!name)
+      .join(', ');
+
```
Comment (torkelo) [inline public/app/features/dashboard/components/PanelEditor/OverrideEditor.tsx:110]: why do we do this (map to just name)? and then map over the properties (the names) and call registry.getIfExists again? why not keep this be the registry items and map over them later. 

So would be jus to filter out the undefined / null ones.

**Comment by dprokop** [comment]
Pushed minor UI change @mckn :

image
image

File: public/app/features/dashboard/components/PanelEditor/OverrideEditor.tsx (hunk 1 above)
Comment (dprokop) [reply public/app/features/dashboard/components/PanelEditor/OverrideEditor.tsx:111]: same question :)

**Review by dprokop — APPROVED** [review]

File: public/app/features/dashboard/components/PanelEditor/OverrideEditor.tsx (hunk 1 above)
Comment (mckn) [reply public/app/features/dashboard/components/PanelEditor/OverrideEditor.tsx:111]: Not really sure I am following. Can you explain a bit more how you think I could change it?

Comment (torkelo) [reply public/app/features/dashboard/components/PanelEditor/OverrideEditor.tsx:111]: Sorry I misread the diff wrong. My comment does not make sense. 

But after reading the code more, It feels like we can get properties and filter out null items once, now it's in two places.  (in renderOverrideTitle and in the body), and two calls to registry.get, think we can simplify the code a bit that way.  

```ts
const configProps =  override.properties
      .map(p => registry.getIfExists(p.id))
      .filter(p => p !== null)
```

**Comment by torkelo** [comment]
Oh shit, maybe this won't need to show the properties, if we did this: https://github.com/grafana/grafana/issues/29895 this should technically not happen

File: public/app/features/dashboard/components/PanelEditor/OverrideEditor.tsx (hunk 1 above)
Comment (mckn) [reply public/app/features/dashboard/components/PanelEditor/OverrideEditor.tsx:111]: Super, I will fix that tomorrow.

**Pull request merged**